pandas/src/period.pyx

```diff
@@ -722,26 +720,31 @@ class Period(object):
             # TODO: Better error message - this is slightly confusing
             raise ValueError('Only mult == 1 supported')
 
-        if self.ordinal is None:
-            self.ordinal = period_ordinal(dt.year, dt.month, dt.day,
+        if ordinal is None:
+            self.ordinal = get_period_ordinal(dt.year, dt.month, dt.day,
                                                 dt.hour, dt.minute, dt.second, dt.microsecond, 0,
                                                 base)
+        else:
+            self.ordinal = ordinal
 
         self.freq = frequencies._get_freq_str(base)
 
-    def __eq__(self, other):
+    def __richcmp__(self, other, op):
         if isinstance(other, Period):
             from pandas.tseries.frequencies import get_freq_code as _gfc
             if other.freq != self.freq:
                 raise ValueError("Cannot compare non-conforming periods")
             if self.ordinal == tslib.iNaT or other.ordinal == tslib.iNaT:
-                return False
-            return (self.ordinal == other.ordinal
+                return _nat_scalar_rules[op]
+            return (PyObject_RichCompareBool(self.ordinal, other.ordinal, op)
                     and _gfc(self.freq) == _gfc(other.freq))
-        return NotImplemented
-
-    def __ne__(self, other):
-        return not self == other
+        else:
+            if op == Py_EQ:
+                return False
+            elif op == Py_NE:
+                return True
+            raise TypeError('Cannot compare type %r with type %r' %
+                            (type(self).__name__, type(other).__name__))
```

Comment:
I guess you can do `NotImplemented` for eq and ne, but not the other comparisons.

Python 2 has weird default behavior where objects of different types have a well defined comparison order instead of raising an exception. This was changed in Python 3:
https://docs.python.org/2/library/stdtypes.html#comparisons
https://docs.python.org/3/library/stdtypes.html#comparisons